src/task/future.rs

```diff
@@ -257,6 +240,15 @@ impl crate::pool::Runner for Runner {
                     task.status.store(COMPLETED, SeqCst);
                     return true;
                 }
+                let extras = { &mut *task.extras.get() };
```

Comment:
I think it should be executed before poll. Otherwise, before the first poll finished, waking a waker moved to other threads will panic.

Reply:
We can use `mut_extras` here.

Reply:
No, it won't. Waking a polling future will mark it NOTIFIED instead of actually waking it up.

Reply:
> We can use mut_extras here

They are not the same `extras`. Besides, task_cell is moved.

Reply:
Ah. I got it.